### DISCUSSION TITLE
Is there an example with Astro?

### DISCUSSION THREAD

**Discussion by iozcelik** [logto-io/logto#4610] [Q&A] [answered]
I am trying to use Logto with Astro framework. I try to use middleware and create some apis. However I stuck and need some help.

Is there any example with Astro? If not, any idea?

**Comment by simeng-li** [comment]
Hi there,

Unfortunately, we don't have an Astro web app integration guide or sample yet.  
Would you mind sharing some more details about your current project status and what kind of problem you have met? 

cc: @wangsijie our web app SDK expert.

**Comment by RyzeKit** [comment, marked as answer]
🚀 There is an example on how to integrate Logto with Astro here: https://github.com/RyzeKit/astro-logto-auth-example

**Reply by IRediTOTO** [reply]
this not cool :s

**Reply by RyzeKit** [reply]
What's not cool about it?

**Reply by IRediTOTO** [reply]
> What's not cool about it?

we must use a 3rd package for that. And it save data to database, that is not purpose of Logto i think. It's just not a best way.

**Reply by RyzeKit** [reply]
👍 Yea, I agree it's not optimal, but this is the best way I can come up with so far. Hopefully someone else can come up with something better 🤞
